Parser/tokenizer.c

```diff
@@ -1245,11 +1250,60 @@ tok_get(struct tok_state *tok, char **p_start, char **p_end)
     /* Set start of current token */
     tok->start = tok->cur - 1;
 
-    /* Skip comment */
+    /* Skip comment, unless it's a type comment */
     if (c == '#') {
+        const char *prefix, *p, *type_start;
+
         while (c != EOF && c != '\n') {
             c = tok_nextc(tok);
         }
+
+        if (tok->type_comments) {
+            p = tok->start;
+            prefix = type_comment_prefix;
+            while (*prefix && p < tok->cur) {
+                if (*prefix == ' ') {
+                    while (*p == ' ' || *p == '\t')
+                        p++;
```

Comment:
I am not sure what are the rules of single line bodies. I would prefer to have them always enclosed in curly braces `{...}`. It looks like you mix both styles. I will label them so that you can deiced what to do.

Reply:
I added a few more braces. I don't know what PEP 7 says, but the code here seems to mostly use the convention that a single `break` or `goto` or `return` doesn't need to be surrounded by braces (except when part of an if-else if-etc. sequence where most other bodies braced). So I added braces to this specific case, but not around several of the others you flagged.